src/components/templates/Poster/Poster.tsx

```diff
@@ -0,0 +1,169 @@
+
+import { IntroVideoPreviewModal } from "./components/IntroVideoPreviewModal";
+
+import "./Poster.scss";
+
+export interface PosterProps {
+  venue: WithId<PosterVenue>;
+}
+
+export const Poster: React.FC<PosterProps> = ({ venue }) => {
+  const [isIntroVideoShown, setIntroVideoShown] = useState<boolean>(false);
+
+  const showIntroVideoModal = () => {
+    setIntroVideoShown(true);
+  };
+
+  const hideIntroVideoModal = () => {
+    setIntroVideoShown(false);
+  };
+
+  const {
+    activeParticipants,
+    passiveListerens,
+
+    isMeActiveParticipant,
+
+    turnVideoOff,
+    turnVideoOn,
+  } = usePosterVideo(venue.id);
+
+  const isPosterLive = venue.isLive;
+
+  const setVenueLiveOn = () => {
+    setVenueLiveStatus(venue.id, true);
+  };
+
+  const setVenueLiveOff = () => {
+    setVenueLiveStatus(venue.id, false);
+  };
+
+  const videoParticipants = activeParticipants.map((participant) => (
+    <VideoParticipant
+      participant={participant}
+      key={participant.identity}
+      additionalClassNames="poster__video-participant"
+    />
+  ));
+
+  const hasFreeSpace = videoParticipants.length < POSTER_CELL_COUNT_MAX;
+
+  return (
+    <>
+      <div className="poster">
+        <div className="poster__header">
+          <div />
+
+          <div className="poster__header--middle-cell">
+            <p className="poster__title">{venue.title}</p>
+            <div className="poster__categories" />
+          </div>
+
+          <div className="poster__header--right-cell">
+            {isMeActiveParticipant && (
+              <div
+                className="poster__control poster__control--stop"
+                onClick={turnVideoOff}
+              >
+                <FontAwesomeIcon icon={faStop} size="lg" />
+                <span className="poster__control-text">Stop video</span>
+              </div>
+            )}
+
+            <OverlayTrigger
+              trigger="click"
+              placement="bottom-end"
+              overlay={
+                <Popover
+                  id="popover-basic"
+                  className="poster__settings-popover"
+                >
+                  <Popover.Content className="poster__settings-popover-content">
+                    <Toggler
+                      type="checkbox"
+                      className="switch-hidden-input"
+                      checked={!!isPosterLive}
+                      onChange={isPosterLive ? setVenueLiveOff : setVenueLiveOn}
+                      title={
+                        isPosterLive ? "Poster is live" : "Make poster live"
+                      }
+                    />
+                  </Popover.Content>
+                </Popover>
+              }
+              rootClose={true}
+            >
+              <div className="poster__control">
+                <FontAwesomeIcon icon={faCog} size="lg" />
+                <span className="poster__control-text">Settings</span>
+              </div>
+            </OverlayTrigger>
```

Comment:
This seems pretty verbose to be included inline here, can we extract it as a smaller sub-component/similar to make this JSX less complex to read through?

Reply:
At the very least the `overlay` part should be extracted.

Reply:
Fixed in https://github.com/sparkletown/sparkle/pull/1299/commits/c23cde4297f7a38d803e0c4f837e99a9c9bdccdd